File: backend/app/schemas/attendance.py
"""Attendance schemas."""

from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class CheckInRequest(BaseModel):
    member_id: int


class CheckOutRequest(BaseModel):
    member_id: int


class AttendanceResponse(BaseModel):
    id: int
    member_id: int
    check_in: datetime
    check_out: Optional[datetime]
    member_name: Optional[str] = None

File: frontend/src/services/attendance.ts
import api from "./api";
import type { Attendance } from "../types";

export const attendanceService = {
  list: () => api.get<Attendance[]>("/attendance/"),
  checkIn: (member_id: number) => api.post<Attendance>("/attendance/checkin", { member_id }),
  checkOut: (member_id: number) => api.post<Attendance>("/attendance/checkout", { member_id }),
};
